brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/spec/VanillaSoftwareProcessLifecycleModifier.java

```diff
@@ -65,96 +74,94 @@ public void apply(EntitySpec<?> entitySpec, NodeTemplate nodeTemplate, Topology
-            applyLifecycle(operations, ToscaNodeLifecycleConstants.CONFIGURE, entitySpec, VanillaSoftwareProcess.CUSTOMIZE_COMMAND, nodeTemplate, topology);
-            applyLifecycle(operations, ToscaNodeLifecycleConstants.START, entitySpec, VanillaSoftwareProcess.LAUNCH_COMMAND, nodeTemplate, topology);
-            applyLifecycle(operations, ToscaNodeLifecycleConstants.STOP, entitySpec, VanillaSoftwareProcess.STOP_COMMAND, nodeTemplate, topology);
-            if (!operations.isEmpty()) {
+        for (String opKey : operations.keySet()) {
+            if (!lifeCycleMapping.containsKey(opKey)) {
                 LOG.warn("Could not translate some operations for " + nodeTemplate + ": " + operations.keySet());
+                continue;
             }
+
+            Operation op = operations.get(opKey);
+            applyLifecycle(op, entitySpec, lifeCycleMapping.get(opKey), nodeTemplate, topology);
         }
     }
 
     protected Map<String, Operation> getStandardInterfaceOperations(NodeTemplate nodeTemplate, Topology topology) {
         Map<String, Operation> operations = MutableMap.of();
-        Interface indexedNodeTemplateInterface = findInterfaceOfNodeTemplate(
-                getIndexedNodeTemplate(nodeTemplate, topology).get().getInterfaces(), VALID_INTERFACE_NAMES);
-        Interface nodeTemplateInterface = findInterfaceOfNodeTemplate(
+        IndexedArtifactToscaElement indexedNodeTemplate = getIndexedNodeTemplate(nodeTemplate, topology).get();
+
+        Interface indexedNodeTemplateInterface = NodeTemplates.findInterfaceOfNodeTemplate(
+                indexedNodeTemplate.getInterfaces(), VALID_INTERFACE_NAMES);
+
+        Interface nodeTemplateInterface = NodeTemplates.findInterfaceOfNodeTemplate(
                 nodeTemplate.getInterfaces(), VALID_INTERFACE_NAMES);
 
-        if ((indexedNodeTemplateInterface != null)
-                && (indexedNodeTemplateInterface.getOperations() != null)) {
+        if (indexedNodeTemplateInterface != null && indexedNodeTemplateInterface.getOperations() != null) {
             operations = MutableMap.copyOf(indexedNodeTemplateInterface.getOperations());
         }
 
-        if ((nodeTemplateInterface != null) && (nodeTemplateInterface.getOperations() != null)) {
-            for (String operationName : nodeTemplateInterface.getOperations().keySet()) {
-                if (operations.containsKey(operationName)) {
-                    operations.put(operationName,
-                            nodeTemplateInterface.getOperations().get(operationName));
-                }
+        if (nodeTemplateInterface != null && nodeTemplateInterface.getOperations() != null) {
+            copyValues(operations, nodeTemplateInterface.getOperations());
+        }
+        return operations;
+    }
+
+    private Map<String, Operation> copyValues(Map<String, Operation> operations, Map<String, Operation> interfaceOperations) {
```

Comment:
I'm lacking some context. Why only copy values from `nodeTemplateInterface` if they are defined in `indexedNodeTemplateInterface`? What difference between the two objects is there in practice? Why not just `operations.putAll(nodeTemplateInterface.getOperations())`?